nixos/file-roller: drop and remove from GNOME module

normalcea opened the pull request:
In conversation with jtojnar, upstream maintainer of file-roller in the NixOS GNOME matrix chat and part of NixOS GNOME team

Wed 03 Sep 2025 02:03:04 PM EDT

>> @normalcea:matrix.org: paging thoughts: I think file roller should be removed from the default install selection since it's not part of GNOME core apps/suite, it's also not seen any new releases outside of bugfix ones and conflicts with Files built-in de-archiving mechanism  

> @jtojnar:matrix.org:  Agreed, I thought I have already removed it. (BTW I am one of the upstream maintainers.)

Removes the file-roller program module from nixos as it's not required for the program to function. This requires adjusting pantheon and cinnamon desktop modules as well as removing file-roller entirely from GNOME as it's not part of the GNOME core applications.

<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform:
  - [ ] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [ ] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [ ] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [x] Module update: when the change is significant.
- [x] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

jtojnar commented on nixos/modules/services/desktop-managers/gnome.nix:
```diff
@@ -482,7 +482,6 @@ in
       # enable that program module if the package hasn't been excluded
       # through `environment.gnome.excludePackages`
       programs.evince.enable = notExcluded pkgs.evince;
-      programs.file-roller.enable = notExcluded pkgs.file-roller;
```
Thanks. I had just this in mind.

I would keep the NixOS module for now since the program contains a D-Bus service, which I believe is preferable to be registered using `services.dbus.packages` (though `environment.systemPackages` also seems sufficient).

normalcea replied:
From what I can tell, packages with dbus services don't need to be explicitly added to `services.dbus.packages` to work since they are wrapped to see their own `$out/share` directory. 

At least in GNOME, most adwaita applications ship their own dbus services and I've never had to set them in `dbus.packages` myself for them to work (`pika-backup` is a good example where I've been able to get automated backups via dbus)

The NixOS module is also a relic of a much older time, it was renamed from the gnome3 module in 2019 and hasn't seen any changes besides being able to override `cfg.package` if we look at the git blame. 

From what I can tell, if file-roller really needs to be placed in `dbus.packages` in other desktops or for an edge use-case it can be placed there by the user or desktop modules can place it there themselves.

normalcea commented:
@jtojnar Have your thoughts changed about removing the module entirely? I'm also okay with removing the module just for GNOME if it turns out the module is necessary for program function on other environments.

jtojnar commented:
Sorry, have not had a time to think about it. Will try to do it after finishing the main GNOME PR.

Commit:
nixos/file-roller: drop

Drops the `programs.file-roller` option as it is not needed for the
program to function and removes all instances of it being used within
nixpkgs.

In addition, file-roller is also removed from being included by
default in GNOME as it's not part of the GNOME core applications and
file archival functions are handled by nautilus (GNOME Files).

jtojnar reviewed (approved):
Thanks, and sorry for the delay.
